internal/tiltfile/version/version.go

```diff
@@ -19,26 +27,50 @@ func (e Extension) NewState() interface{} {
 	}
 }
 
-func (Extension) OnStart(env *starkit.Environment) error {
-	return env.AddBuiltin("version_settings", setVersionSettings)
+func (e Extension) OnStart(env *starkit.Environment) error {
+	return env.AddBuiltin("version_settings", e.setVersionSettings)
 }
 
-func setVersionSettings(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
-	var checkUpdates bool
+func (e Extension) setVersionSettings(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	// it's awkward to tell from UnpackArgs whether an optional value was set, so let's just make sure it has
+	// the right initial value instead
+	m, err := starkit.ModelFromThread(thread)
+	if err != nil {
+		return nil, errors.Wrap(err, "internal error reading version settings from starkit model")
+	}
+	state, err := GetState(m)
+	if err != nil {
+		return nil, errors.Wrap(err, "internal error reading version settings from starkit model")
+	}
+
+	checkUpdates := state.CheckUpdates
+	var constraint string
 	if err := starkit.UnpackArgs(thread, fn.Name(), args, kwargs,
-		"check_updates", &checkUpdates); err != nil {
+		"check_updates?", &checkUpdates,
```

Comment:
i'm a little bit worried about every extension inventing a new convention for unpacking args into a data model. If there's a shortcoming in the API that we want to fix, lots of bespoke call patterns make it harder to fix.

Reply:
i wonder if a pattern that might make sense is to do the UnpackArgs in SetState()?

e.g.,

```
err = starkit.SetState(thread, func(settings model.VersionSettings) (model.VersionSettings, err) {
  ...
  starkit.UnpackArgs(thread, fn.Name(), args, kwargs,
		"check_updates", &settings.CheckUpdates)
  ...
}
```

not totally sure, just spitballing here, this is ok for now

Reply:
yeah, that's a lot simpler, thanks

I also started a [convo](https://github.com/google/starlark-go/issues/267) on the starlark-go repo that I think will let us cheaply deal with types like `Union[bool|None]` that have been annoying in a bunch of cases.